File: app/src/main/java/uo206367/dspm/miw/dsdmchatt/model/Message.java
package uo206367.dspm.miw.dsdmchatt.model;

public class Message {

    private String section;
    private Data data;

    public Message(String section, Data data) {
        this.section = section;
        this.data = data;
    }

    public Message(String userName, String text){
        setSection("messages");
        setData(new Data(userName, text));
    }

    public String getSection() {
        return section;
    }

    public void setSection(String section) {
        this.section = section;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }
}

File: app/src/main/java/uo206367/dspm/miw/dsdmchatt/model/Message.kt
package uo206367.dspm.miw.dsdmchatt.model

import uo206367.dspm.miw.dsdmchatt.R

class Message {

    var section: String? = null
    var data: Data? = null

    constructor(section: String, data: Data) {
        this.section = section
        this.data = data
    }

    constructor(userName: String, text: String) {
        section = "messages"
        data = Data(userName, text)
    }

    fun printMessage(msg: String?): String {
        when (section) {
            "messages" -> return data!!.userName + ":\n\t\t" + data!!.text

            "people" -> return java.lang.String.format(msg, data!!.userName)

            else -> return ""
        }
    }
}
